### PR TITLE
Added public API of UI Grid

### PR DESCRIPTION
**Pull request by pavel-trafimuk**
UI Grid Builders are still not ready
Defaults values are not ready

### CODE REVIEW

**Comment by pavel-trafimuk** [comment]
@ bigsnickers

File: Sources/ViberUIGridView/String+Html.swift [hunk 1]
```diff
@@ -0,0 +1,32 @@
+//
+//  File.swift
```
Comment (bigsnickers) [inline Sources/ViberUIGridView/String+Html.swift]: Please fix the file header.

File: Sources/ViberUIGridView/String+Markdown.swift [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+//  
+//
+//  Created by Pavel Trafimuk on 13/01/2023.
+//  Viber Media, Inc.
+
+import Foundation
+
+// Markdown supported only in text/picture messages
+extension String {
+    public func bold() -> String {
```
Comment (bigsnickers) [inline Sources/ViberUIGridView/String+Markdown.swift]: Please rename to `markdownBold()`, similar as you did with `htmlBold()`

File: Sources/ViberUIGridView/UIGrid.swift [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+                                                             actionType: .none,
+                                                             isSilent: false)],
+                                             buttonsGroupColumns: nil,
+                                             buttonsGroupRows: nil,
+                                             heightScale: nil,
+                                             backgroundColor: nil,
+                                             inputFieldState: nil,
+                                             favoritesMetadata: nil)
+    
+    public static func keyboard(with buttons: [UIGridButton],
```
Comment (bigsnickers) [inline Sources/ViberUIGridView/UIGrid.swift]: `forKeyboard(...)` for readability

File: Sources/ViberUIGridView/UIGrid.swift [hunk 4]
```diff
@@ -0,0 +1,241 @@
...
+                      buttons: buttons,
+                      buttonsGroupColumns: buttonsGroupColumns,
+                      buttonsGroupRows: buttonsGroupRows,
+                      heightScale: nil,
+                      backgroundColor: backgroundColor,
+                      inputFieldState: inputFieldState,
+                      favoritesMetadata: nil)
+    }
+    
+    public static func rich(with buttons: [UIGridButton],
```
Comment (bigsnickers) [inline Sources/ViberUIGridView/UIGrid.swift]: rich what?

File: Sources/ViberUIGridView/UIGrid.swift [hunk 5]
```diff
@@ -0,0 +1,241 @@
...
+        self.buttons = buttons
+        self.buttonsGroupColumns = try container.decodeIfPresent(Int.self, forKey: .buttonsGroupColumns)
+        self.buttonsGroupRows = try container.decodeIfPresent(Int.self, forKey: .buttonsGroupRows)
+        self.heightScale = try container.decodeIfPresent(Int.self, forKey: .heightScale)
+        self.backgroundColor = try container.decodeIfPresent(String.self, forKey: .backgroundColor)
+        self.inputFieldState = try container.decodeIfPresent(UIGrid.InputFieldState.self, forKey: .inputFieldState)
+        self.favoritesMetadata = try container.decodeIfPresent(UIGridFavorites.self, forKey: .favoritesMetadata)
+    }
+    
+//    public static let keyboard(buttonBuilders: [UIGridButtonBuilder?],
```
Comment (bigsnickers) [inline Sources/ViberUIGridView/UIGrid.swift:188]: Either place a TODO what should be done to uncomment this code or simply remove it

File: Sources/ViberUIGridView/UIGridBuilder.swift [hunk 6]
```diff
@@ -0,0 +1,252 @@
+////
```
Comment (bigsnickers) [inline Sources/ViberUIGridView/UIGridBuilder.swift:1]: What is here?

File: Sources/ViberUIGridView/UIGridButton.swift [hunk 7]
```diff
@@ -0,0 +1,398 @@
+//
+//  File.swift
+//  
+//
+//  Created by Pavel Trafimuk on 13/07/2023.
+//
+
+import Foundation
+
+// TODO: WARN add
```
Comment (bigsnickers) [inline Sources/ViberUIGridView/UIGridButton.swift:8]: add what?

File: Sources/ViberUIGridView/UIGridButtonMap.swift [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+    }
+    
+    /// Location longitude (format: "3.12345")
+    public let longitudeString: String
+
+    public var longitude: Double? {
+        Double(longitudeString)
+    }
+    
+    public enum CodingKeys: String, CodingKey {
```
Comment (bigsnickers) [inline Sources/ViberUIGridView/UIGridButtonMap.swift]: private

File: Sources/ViberUIGridView/UIGridInternalBrowser.swift [hunk 9]
```diff
@@ -0,0 +1,122 @@
...
+//
+
+import Foundation
+
+/// JSON Object, which includes internal browser configuration for OpenURL action with "internal" type. Will be ignored if action is not openUrl in internal browser.
+/// API: rev.3+
+public struct UIGridInternalBrowser: Codable, Equatable {
+    
+    public enum ActionButton: String, Codable, Equatable, UnknownDecodable {
+        public static var unknown = ActionButton.none
```
Comment (bigsnickers) [inline Sources/ViberUIGridView/UIGridInternalBrowser.swift]: either let or make it computed

File: Sources/ViberUIGridView/UIGridType.swift [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+    public func defaultButtonSize(groupSize: UIGridSize) -> UIGridSize {
+        switch self {
+        case .keyboard: return .init(columns: groupSize.columns, rows: 1)
+        case .richMedia: return groupSize
+            
+        case .unknown: return groupSize
+        }
+    }
+    
+//    public func isValid(actionType: UIGridButton.ActionType) -> Bool {
```
Comment (bigsnickers) [inline Sources/ViberUIGridView/UIGridType.swift:33]: why commented?

File: Tests/ViberUIGridViewTests/jsons/botKeyboardCyrillic.json [hunk 11]
```diff
@@ -0,0 +1,87 @@
+{
+   "Type":"keyboard",
+   "DefaultHeight":true,
+   "Buttons":[
+      {
+         "ActionType":"reply",
+         "ActionBody":"Где получить полную и достоверную информацию о коронавирусе?",
```
Comment (bigsnickers) [inline Tests/ViberUIGridViewTests/jsons/botKeyboardCyrillic.json]: Pls remove russian language. It is toxic these days

File: Tests/ViberUIGridViewTests/jsons/botKeyboardInvalidJSON.json [hunk 12]
```diff
@@ -0,0 +1,26 @@
...
+                "TextHAlign" : "left",
+                "TextSize": "large"
+                },
+                {
+                "Columns" : 4,
+                "Rows" : 2,
+                "BgColor": "#FFFFFF",
+                "ActionType": "open-url",
+                "ActionBody": "http://bbc.com/trump.html",
+                "Text": "<i><font color=\"#bfbfbf\">CNN@Donald Trump met Wednesday with GOP strategist Paul... Republican presidential</font> <br><font color=\"#FF0000\">Read More...</font></i>",
```
Comment (bigsnickers) [inline Tests/ViberUIGridViewTests/jsons/botKeyboardInvalidJSON.json]: same as Trump. It's Biden time

**Comment by bigsnickers** [comment]
Reviewed. Some of the comments I've written only once. Please check similar places In other files.

**Comment by pavel-trafimuk** [comment]
All fixes, except notes, which partially done in the next PR

**Pull request merged**